File: NYPC-2018/day-1/prob8.py
import sys
sys.setrecursionlimit(999999999)

x, y = input().split()
x = int(x)
y = int(y)

def btn(a, b):
    return a + b

def rec(a, b):
    if a > x or b > y: return
    if (a, b) == (x, y):
        print('POSSIBLE')
        exit()
    rec(btn(a, b), b)
    rec(a, btn(a, b))

rec(1, 1)
if (x, y) == (1, 0):
    print('POSSIBLE')
    exit()
print('IMPOSSIBLE')

# 이거 했는데 시간초과? 였나 나서 C로 짜서 제출해서 50점 맞은 것 같은데 C 소스 실수로 날려먹음

File: NYPC-2018/day-1/prob8.java
import java.util.Scanner;

public class test8 {
    
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);
        int x = scn.nextInt();
        int y = scn.nextInt();
        rec(x, y, 1, 1);
        if (x == 1 && y == 0){
            System.out.println("POSSIBLE");
            System.exit(0);
        }
        System.out.println("IMPOSSIBLE");
    }

    public static void rec (int x, int y, int a, int b) {
        if (a > x || b > y)
            return;
        if (a == x && b == y){
            System.out.println("POSSIBLE");
            System.exit(0);
        }
        rec(x, y, btn(a, b), b);
        rec(x, y, a, btn(a, b));
    }
    
    public static int btn (int a, int b) {
        return a + b;
    }

}
 